src/sentry/demo/data_population.py

```diff
@@ -1,18 +1,133 @@
-from sentry.utils.samples import generate_user
-from sentry.utils.samples import create_sample_event
+import logging
+import functools
+import random
+from typing import (
+    List,
+    Callable,
+)
 
+from datetime import timedelta
+from django.utils import timezone
+from sentry.interfaces.user import User as UserInterface
+from sentry.models import Project
+from sentry.utils import json
+from sentry.utils.samples import random_geo, random_ip, create_sample_event
+from sentry.utils.snuba import SnubaError
 
-def populate_python_project(project):
-    create_sample_event(
-        project=project,
-        platform="python",
-        user=generate_user(),
+
+MAX_DAYS = 2
+SCALE_FACTOR = 0.5
+BASE_OFFSET = 0.5
+NAME_STEP_SIZE = 20
+
+
+logger = logging.getLogger(__name__)
+
+
+def distribution_v1(hour: int) -> int:
+    if hour > 9 and hour < 12:
+        return 8
+    if hour > 6 and hour < 15:
+        return 3
+    if hour > 4 and hour < 20:
+        return 2
+    return 1
+
+
+def distribution_v2(hour: int) -> int:
+    if hour > 18 and hour < 20:
+        return 22
+    if hour > 9 and hour < 14:
+        return 7
+    if hour > 3 and hour < 22:
+        return 4
+    return 2
+
+
+def distribution_v3(hour: int) -> int:
+    if hour > 21:
+        return 11
+    if hour > 6 and hour < 15:
+        return 6
+    if hour > 3:
+        return 2
+    return 1
+
+
+distrubtion_fns = [distribution_v1, distribution_v2, distribution_v3]
+
+
+@functools.lru_cache(maxsize=None)
+def get_list_of_names() -> List[str]:
+    with open("src/sentry/demo/data/names.json") as f:
+        return json.load(f)
+
+
+def generate_user():
+    name_list = get_list_of_names()
+    id_0_offset = random.randrange(0, len(name_list), NAME_STEP_SIZE)
+    name = name_list[id_0_offset]
+    email = f"{name.lower()}@example.com"
+    return UserInterface.to_python(
+        {
+            "id": id_0_offset + 1,
+            "email": email,
+            "ip_address": random_ip(),
```

Comment:
Could we reuse caching here to get consistent ip&geo based on the id?

Reply:
@wmak good point! I didn't think of that. Cacheing is a good idea.